plugins/storage/volume/datera/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/DateraPrimaryDataStoreLifeCycle.java

```diff
@@ -395,6 +395,15 @@ public void disableStoragePool(DataStore dataStore) {
         dataStoreHelper.disable(dataStore);
     }
 
+    @Override
+    public void changeStoragePoolScopeToZone(DataStore store, ClusterScope clusterScope, HypervisorType hypervisorType) {
+        /*
+         * We need to attach all VMware, Xenserver and KVM hosts in the zone.
+         * So pass hypervisorTpe as null.
+         */
+        super.changeStoragePoolScopeToZone(store, clusterScope, null);
```

Comment:
is this feature enabled for Detera storage ?
or this will throw an exception ?

Reply:
Feature is not tested and supported on Datera. But if a root admin wishes, he can use it. We are not preventing any kind of storage from using this feature. The feature will make the DB updates and connect/disconnect the hosts to the storage pool as required. It is not ensured that it will work with configurations other than NFS/Ceph, KVM/VMware and Default Primary storage provider.
Have mentioned this as a warning in UI and Api docs.

Reply:
> Feature is not tested and supported on Datera. But if a root admin wishes, he can use it. We are not preventing any kind of storage from using this feature. The feature will make the DB updates and connect/disconnect the hosts to the storage pool as required. It is not ensured that it will work with configurations other than NFS/Ceph, KVM/VMware and Default Primary storage provider.
> Have mentioned this as a warning in UI and Api docs.

Fair enough
Thanks